linker/Linker.Steps/MarkStep.cs

```diff
@@ -273,6 +282,127 @@ void MarkCustomAttributes (ICustomAttributeProvider provider)
 			}
 		}
 
+		void MarkUserDependency (MethodReference context, CustomAttribute ca)
+		{
+			if (ca.ConstructorArguments.Count == 1 && ca.ConstructorArguments[0].Value is string s) {
+				string member = null;
+				string type = null;
+				string[] signature = null;
+				TypeDefinition td = null;
+
+				var sign = s.IndexOf ('(');
+				if (sign > 0) {
+					var parameters = s.Substring (sign + 1, s.Length - sign - 2);
+					signature = string.IsNullOrEmpty (parameters) ? Array.Empty<string> () : parameters.Split (',');
+					var idx = s.LastIndexOf ('.', sign);
+					if (idx > 0) {
+						member = s.Substring (idx + 1, sign - idx - 1).TrimEnd ();
+						type = s.Substring (0, idx);
+					} else {
+						member = s.Substring (0, sign - 1);
+						td = context.DeclaringType.Resolve ();
+					}
+				} else {
+					var idx = s.LastIndexOf ('.');
+					if (idx > 0) {
+						member = s.Substring (idx + 1);
+						type = s.Substring (0, idx);
+					} else {
+						member = s;
+						td = context.DeclaringType.Resolve ();
+					}
+				}
+
+				if (td == null) {
```

Comment:
Can you put handling of the not found case into something we can override?  I strongly suspect that even if it's decided monolinker won't fail on types that cannot be found, we will want UnityLinker to fail when types/methods/fields are not found.

Reply:
I'll leave that like that for now as we want to experiment of making it an error by default